lib/vfs/src/lib.rs

```diff
@@ -0,0 +1,443 @@
+use std::any::Any;
+use std::ffi::OsString;
+use std::fmt;
+use std::io::{self, Read, Seek, Write};
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use tracing::debug;
+
+#[cfg(feature = "host_fs")]
+pub mod host_fs;
+//pub mod vfs_fs;
+#[cfg(feature = "mem_fs")]
+pub mod mem_fs;
```

Comment:
I think both features are mutually exclusive.

We should add something like this:

```rust
#[cfg(all(feature = "host_fs", feature = "mem_fs"))]
compile_error!("…");
```

Thoughts?

Reply:
See https://github.com/wasmerio/wasmer/pull/2523 to solve this comment.